systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java

```diff
@@ -1069,6 +1075,499 @@ void testRestoreOffsetsInConsumerGroup(ExtensionContext extensionContext) {
+            topicTestName2, kafkaClusterTargetName, userTarget.getMetadata().getName());
+
+        internalKafkaClient = internalKafkaClient.toBuilder()
+            .withClusterName(kafkaClusterTargetName)
+            .withTopicName(topicTestName2)
+            .withKafkaUsername(userTarget.getMetadata().getName())
+            .build();
+
+        LOGGER.info("Sending messages to - topic {}, cluster {} and message count of {}",
+            topicTestName2, kafkaClusterTargetName, messagesCount);
+        internalKafkaClient.checkProducedAndConsumedMessages(internalKafkaClient.sendMessagesTls(), internalKafkaClient.receiveMessagesTls());
+
+        // Initialize CertSecretSource with certificate and secret names for source
+        CertSecretSource certSecretSource = new CertSecretSource();
+        certSecretSource.setCertificate("ca.crt");
+        certSecretSource.setSecretName(KafkaResources.clusterCaCertificateSecretName(kafkaClusterSourceName));
+
+        // Initialize CertSecretSource with certificate and secret names for target
+        CertSecretSource certSecretTarget = new CertSecretSource();
+        certSecretTarget.setCertificate("ca.crt");
+        certSecretTarget.setSecretName(KafkaResources.clusterCaCertificateSecretName(kafkaClusterTargetName));
+
+        // Deploy Mirror Maker 2.0 with tls listener and mutual tls auth
+        KafkaMirrorMaker2ClusterSpec sourceClusterWithTlsAuth = new KafkaMirrorMaker2ClusterSpecBuilder()
+            .withAlias(kafkaClusterSourceName)
+            .withBootstrapServers(KafkaResources.tlsBootstrapAddress(kafkaClusterSourceName))
+            .withNewKafkaClientAuthenticationTls()
+                .withNewCertificateAndKey()
+                    .withSecretName(kafkaUserSourceName)
+                    .withCertificate("user.crt")
+                    .withKey("user.key")
+                .endCertificateAndKey()
+            .endKafkaClientAuthenticationTls()
+            .withNewTls()
+                .withTrustedCertificates(certSecretSource)
+            .endTls()
+            .build();
+
+        KafkaMirrorMaker2ClusterSpec targetClusterWithTlsAuth = new KafkaMirrorMaker2ClusterSpecBuilder()
+            .withAlias(kafkaClusterTargetName)
+            .withBootstrapServers(KafkaResources.tlsBootstrapAddress(kafkaClusterTargetName))
+            .withNewKafkaClientAuthenticationTls()
+                .withNewCertificateAndKey()
+                    .withSecretName(kafkaUserTargetName)
+                    .withCertificate("user.crt")
+                    .withKey("user.key")
+                .endCertificateAndKey()
+            .endKafkaClientAuthenticationTls()
+            .withNewTls()
+                .withTrustedCertificates(certSecretTarget)
+            .endTls()
+            .addToConfig("config.storage.replication.factor", -1)
+            .addToConfig("offset.storage.replication.factor", -1)
+            .addToConfig("status.storage.replication.factor", -1)
```

Comment:
Maybe I'm wrong, but this should be there by default? Or why do you have it here but not in other specs?

Reply:
Yeah, or `1`. I need to doublecheck but will remove it once I know correct value :-)